proxy/http2/Http2ConnectionState.cc

```diff
@@ -898,17 +900,11 @@ Http2ConnectionState::main_event_handler(int event, void *edata)
                 error.msg);
         }
         this->send_goaway_frame(this->latest_streamid_in, error.code);
+        this->ua_session->set_half_close_flag(true);
```

Comment:
I think ua_session's mutex is already held. Because here is always on call stack from ua_session.